CoreSchemaView/src/au/gov/asd/tac/constellation/views/schemaview/providers/TransactionTypeNodeProvider.java

```diff
@@ -180,54 +180,25 @@ private void populateTree() {
 
     private boolean isFilterMatchCurrentNode(SchemaTransactionType treeItem) {
-        boolean found = isFilterMatchText(treeItem.getName());
-        if (!found) {
-            found = isFilterMatchAnyProperty(treeItem);
-        }
-        return found;
+        return filterText.getText().toLowerCase().isEmpty() ? false
+                : isFilterMatchText(treeItem.getName()) || isFilterMatchAnyProperty(treeItem);
     }
 
     private boolean isFilterMatchAnyProperty(SchemaTransactionType treeItem) {
-        boolean found = false;
-        final String name = treeItem.getName();
-        found = isFilterMatchText(name);
-        if (!found) {
-            final String description = treeItem.getDescription();
-            found = isFilterMatchText(description);
-        }
-        if (!found) {
-            final String color = treeItem.getColor().toString();
-            found = isFilterMatchText(color);
-        }
-        if (!found) {
-            final String style = treeItem.getStyle().toString();
-            found = isFilterMatchText(style);
-        }
-        if (!found) {
-            final String directedLabel = treeItem.isDirected().toString();
-            found = isFilterMatchText(directedLabel);
-        }
-        if (!found) {
-            final String hierachy = treeItem.getHierachy();
-            found = isFilterMatchText(hierachy);
-        }
-        if (!found) {
-            for (String property : treeItem.getProperties().keySet()) {
-                final Object propertyValue = treeItem.getProperty(property);
-                if (propertyValue != null) {
-                    found = isFilterMatchText(propertyValue.toString());
-                    if (found) {
-                        break;
-                    }
-                }
-            }
-        }
-        return found;
+        return isFilterMatchText(treeItem.getName())
+                || isFilterMatchText(treeItem.getDescription())
+                || isFilterMatchText(Objects.toString(treeItem.getColor().toString(), ""))
+                || isFilterMatchText(Objects.toString(treeItem.getStyle().toString(), ""))
+                || isFilterMatchText(Objects.toString(treeItem.isDirected().toString(), ""))
+                || isFilterMatchText(treeItem.getHierachy())
+                || !(treeItem.getProperties().keySet().isEmpty()) && treeItem.getProperties().keySet().stream().anyMatch(property
+                -> property != null && isFilterMatchText(property.toString())
+        );
     }
 
     private boolean isFilterMatchText(final String propertyValue) {
         final String filterInputText = filterText.getText().toLowerCase();
-        return filterInputText.isEmpty() ? false : startsWithRb.isSelected()
+        return (filterInputText.isEmpty() || propertyValue.isEmpty()) ? false : startsWithRb.isSelected()
                 ? propertyValue.toLowerCase().startsWith(filterInputText) : propertyValue.toLowerCase().contains(filterInputText);
```

Comment:
change this return to `(!filterInputText.isEmpty() && !propertyValue.isEmpty()) || (startsWithRb.isSelected()
                ? propertyValue.toLowerCase().startsWith(filterInputText) : propertyValue.toLowerCase().contains(filterInputText))`

Reply:
Same issue as above.

Reply:
see above comment

Reply:
Similarly,  && seems to work here.